src/sql/workbench/contrib/dashboard/browser/dashboardActions.ts

```diff
@@ -100,6 +104,9 @@ export class OEManageConnectionAction extends Action {
 				}
 			}
 		}
+		else if (actionContext === undefined || actionContext === null) {
+			connectionProfile = TaskUtilities.getCurrentGlobalConnection(this._objectExplorerService, this._connectionManagementService, this._editorService) as ConnectionProfile;
```

Comment:
You should avoid using `as` whenever possible - since that's just overriding the typing that TS provides and generally indicates an error. IConnectionProfile !== ConnectionProfile, and casting it like this is dangerous (what happens if someone tries to call a function that ConnectionProfile defines such as `matches`?).

Instead, you should be doing something like this 

```suggestion

			const globalProfile = TaskUtilities.getCurrentGlobalConnection(this._objectExplorerService, this._connectionManagementService, this._editorService);
			connectionProfile = globalProfile ? ConnectionProfile.fromIConnectionProfile(this._capabilitiesService, globalProfile) : undefined;
```